src/coreComponents/codingUtilities/StringUtilities.hpp

```diff
@@ -60,6 +60,16 @@ string join( IT first, IT last, S const & delim = S())
   return oss.str();
 }
 
+template< typename S = char, typename T, typename ... Ts >
+string concat( S const & delim, T const & v, Ts const & ... vs )
+{
+  std::ostringstream oss;
+  oss << v;
+  using expander = int[];
+  (void) expander{ 0, ( void (oss << delim << vs ), 0 )... };
```

Comment:
Ah yes, I lifted this "fold expression" technique from LvArray:
https://github.com/GEOSX/LvArray/blob/b80e1b1df9ad2d3e9163d7a5ff088f0e3895ebb8/src/indexing.hpp#L141-L152
(I couldn't use that function directly because it enforces a slightly different format than what I needed).

The leading 0 is needed to account for the case where the variadic pack is empty (0-element arrays are not allowed). This case is currently not triggered anywhere, but it could be in the future.

The `void` cast is indeed to silence warnings, and it seems like it cannot be well-formed without a separate `using` declaration.

Reply:
> The leading 0 is needed to account for the case where the variadic pack is empty (0-element arrays are not allowed). This case is currently not triggered anywhere, but it could be in the future.

That's what I was thinking and I actually tried on godbolt.org with multiple compilers.
Getting rid of the leading `0` works smoothly.
So I was wondering if I was missing something. 🤷 

My guess would be that the initial _stackoverflow_ answer gets the extra zero before it did invade the world! 🤣 🚀

Reply:
It works by default, but not with `-pedantic`: https://godbolt.org/z/hsxnzrrPs
Also, the standard says "N shall be greater than zero": http://eel.is/c++draft/dcl.array#1
So I'll keep it to be on the safe side :safety_vest:

Reply:
OK, this is precise!